### PR TITLE
Contribution documentation improvements

### PR DESCRIPTION
**Pull request by nirname**
## :bookmark_tabs: Summary

Brief description about the content of your PR.

Related #2910

## :straight_ruler: Design Decisions

* Tutorials moved to ecosystem because they are seemed not to be not officially supported
* All the pages `development.md`, `docker-development.md`, `code.md` and `documentation.md` are combined together and refined, and named `contributing.md`. 
* `CONTRIBUTING.md` is now a symlink to our documentation
* Added introduction page for contributing based on #866 so as to purify contributing guide and describe other ways of participating in project
* Link to contributing removed from versions sub-menu, because it is already there at the top bar
* I noticed that in many cases we used right sidebar with all the heading included. Now displaying all the levels of heading at the right sidebar is the default behaviour. This will make access to the sections easier and enforce better planning while writing documentation so as to avoid extra headings.
* I am not sure how many people are using GitHub directly to contribute to documentation, but I think that keeping section on "how to use Github features" is an extra for our contributing to documentation guidelines. Anyway, the overall contributing process is the same regardless of what a developer prefer - GitHub UI vs his favorite editor
* Removed recommendation to contribute directly to development branch for docs and opening PR-s targeting master branch. My thoughts that these exceptions only make process more complicated. Better keep it uniform
* Documentation section is partially intersects with #4899 and that is why it was taken as a reference in some sense
* Advanced section of configuration was removed because it was empty. Some TODO's are removed as well, because they looked messy

### :clipboard: Tasks

Make sure you

- [x] :book: have read the [contribution guidelines](https://github.com/mermaid-js/mermaid/blob/develop/CONTRIBUTING.md)
- [x] :computer: have added necessary unit/e2e tests.
- [x] :notebook: have added documentation. Make sure [`MERMAID_RELEASE_VERSION`](https://github.com/mermaid-js/mermaid/blob/develop/packages/mermaid/src/docs/community/development.md#3-update-documentation) is used for all new features.
- [x] :bookmark: targeted `develop` branch

### CODE REVIEW

File: packages/mermaid/src/docs/community/contributing.md [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+### Update Documentation
+
+```tip
+Our documentation is managed in `packages/mermaid/src/docs`. Details on how to edit is in the [documentation section](#contributing-documentation)
+```
+
+If the users have no way to know that things have changed, then you haven't really _fixed_ anything for the users; you've just added to making Mermaid feel broken.
+Likewise, if users don't know that there is a new feature that you've implemented, it will forever remain unknown and unused.
+
+The documentation has to be updated to users know that things have changed and added!
```
Comment (sidharthv96) [inline packages/mermaid/src/docs/community/contributing.md:364]:
```suggestion
The documentation has to be updated for users to know that things have been changed or added!
```

**Review by sidharthv96 — COMMENTED** [review]
Awesome work @nirname, just some minor changes needed!

**Comment by nirname** [comment]
@sidharthv96 Thanks, nice review, I'll fix it as soon as possible

File: packages/mermaid/src/docs/community/contributing.md [hunk 2]
```diff
@@ -0,0 +1,514 @@
...
+We support **development within Docker** environment along with **host setup**. You may choose it up to your preferences.
+
+**Host**
+
+These are the tools we use for working with the code and documentation:
+
+- [volta](https://volta.sh/) to manage node versions.
+- [Node.js](https://nodejs.org/en/). `volta install node`
+- [pnpm](https://pnpm.io/) package manager. `volta install pnpm`
+- [npx](https://docs.npmjs.com/cli/v8/commands/npx) the packaged executor in npm. This is needed [to install pnpm.](#install-packages)
```
[the remark this answers is not in the document]
Comment (nirname) [reply packages/mermaid/src/docs/community/contributing.md:51]: Do we still need volta?

**Comment by nirname** [comment]
@sidharthv96 fixed everything, unless I had missed smth. I have a question. Some of our docs do not need mermaid code to be compiled and embedded on the page along with diagram, because diagram is only for description of the document not for description of the syntax. Thus I used `mermaid-nocode` keyword. But GitHub does not support that, so when using diagram solely for illustrative purposes I face a dilemma:

* avoid diagram code embedded in the doc but sacrifice GitHub contributing guidelines appearance
* put up with the extra code on the documentation website but support those who are using GitHub UI
image

Ideally I would support both, but then we have to change `MERMAID_EXAMPLE_KEYWORDS` to something else. And that will influence other parts of documentation as well. So I decided to tackle this problem separately and after a discussion. Right now I can get rid of `mermaid-nocode` in my examples if needed.

File: packages/mermaid/src/docs/community/contributing.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (sidharthv96) [reply packages/mermaid/src/docs/community/contributing.md]: Nope. Just pnpm, which can be used to install node.

**Comment by sidharthv96** [comment]
In this specific case, I'd prioritize the website user over the GitHub one. Although, the website user seeing the code might not be a major issue. 

Wonder if this could help. If the markdown parser gives us the full text, that would fix our problem. 

````

```mermaid nocode
flowchart
a --> b
```


````

File: packages/mermaid/src/docs/community/contributing.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (nirname) [reply packages/mermaid/src/docs/community/contributing.md]: Removed `volta` requirement completely, as well as `npx` and updated docker related files as well accordingly.

**Comment by nirname** [comment]
I think we will leave `mermaid-nocode` for now, not a big deal if someone won't see very simple and not really important diagram on GitHub directly. Lets address those keywords separately

**Comment by nirname** [comment]
It's unclear why deployment failed
```
1:10:20 AM: + rm -rf mermaid-live-editor
1:10:21 AM: + git clone --single-branch https://github.com/mermaid-js/mermaid-live-editor.git
1:10:21 AM: Cloning into "mermaid-live-editor"...
1:10:24 AM: + cd mermaid-live-editor
1:10:24 AM: + npm install
1:10:35 AM: Failed during stage "building site": Build script returned non-zero exit code: 2
1:10:35 AM: npm ERR! code ERESOLVE
1:10:35 AM: npm ERR! ERESOLVE unable to resolve dependency tree
1:10:35 AM: npm ERR!
1:10:35 AM: npm ERR! While resolving: mermaid-live-editor@2.0.67
1:10:35 AM: npm ERR! Found: svelte@4.2.8
1:10:35 AM: npm ERR! node_modules/svelte
1:10:35 AM: npm ERR!   dev svelte@"4.2.8" from the root project
1:10:35 AM: npm ERR!
1:10:35 AM: npm ERR! Could not resolve dependency:
1:10:35 AM: npm ERR! peer svelte@"^3.2.0" from eslint-plugin-svelte3@4.0.0
1:10:35 AM: npm ERR! node_modules/eslint-plugin-svelte3
1:10:35 AM: npm ERR!   dev eslint-plugin-svelte3@"4.0.0" from the root project
1:10:35 AM: npm ERR!
1:10:35 AM: npm ERR! Fix the upstream dependency conflict, or retry
1:10:35 AM: npm ERR! this command with --force or --legacy-peer-deps
1:10:35 AM: npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
1:10:35 AM: npm ERR!
1:10:35 AM: npm ERR!
1:10:35 AM: npm ERR! For a full report see:
1:10:35 AM: npm ERR! /opt/buildhome/.npm/_logs/2024-01-13T22_10_24_182Z-eresolve-report.txt
1:10:35 AM: npm ERR! A complete log of this run can be found in: /opt/buildhome/.npm/_logs/2024-01-13T22_10_24_182Z-debug-0.log
```
Is this step required for contributing to documentation?

File: packages/mermaid/src/docs/community/contributing.md [hunk 3]
```diff
@@ -0,0 +1,523 @@
...
+**Host**
+
+```bash
+pnpm run dev
+```
+
+**Docker**
+
+```bash
+./run dev
```
Comment (sidharthv96) [inline packages/mermaid/src/docs/community/contributing.md:254]:
```suggestion
./run pnpm dev
```

**Comment by sidharthv96** [comment]
I have fixed the build.

**Comment by nirname** [comment]
@sidharthv96 updated branch so build works now

**Review by jgreywolf — APPROVED** [review]

**Comment by nirname** [comment]
@aloisklink I remember you were asking to make CONTRIBUTING.md to be exactly the same as our website contributing documentation and combine everything in one page. This PR does exactly what you were looking for

**Comment by huynhicode** [comment]
@nirname Great job

Perhaps we can remove the "Contributing" link from the dropdown since there is a link in the Navbar.

Screenshot 2024-01-18 at 7 53 32 AM

**Comment by nirname** [comment]
@huynhicode Yep, already removed

**Pull request merged**